ring/partition_ring.go

```diff
@@ -87,6 +91,19 @@ func (r *PartitionRing) ActivePartitionForKey(key uint32) (int32, error) {
 	return 0, ErrNoActivePartitionFound
 }
 
+// ShuffleShardSize returns number of partitions that would be in the result of ShuffleShard call with the same size.
+func (r *PartitionRing) ShuffleShardSize(size int) int {
+	if size <= 0 || size > r.activePartitionsCount {
+		return r.activePartitionsCount
+	}
+
+	// return min(size, r.activePartitionsCount)
```

Comment:
[nit] I would remove this line.

Reply:
Done. I left it there as a version to use after we switch to Go 1.21 :)